### ISSUE TITLE
calling JS files in /static folder by script tag don't get fired on develop mode

### ISSUE THREAD

**Issue by github0013** [gatsbyjs/gatsby#8755]
## Description

Calling a JS file with e.g. `<script src="/static/folder/file.js" />` on **gatsby develop** doesn't call the file. (it doesn't show up on network tab either)

However, `gatsby build` then `gatsby serve` will call the `<script />` tag.

### Steps to reproduce

https://github.com/github0013/gatsby_script_tag
I created a repo.  check `/static/test.js` and `/src/pages/index.js`

### Expected result

`<script />` tag should be called on either `develop` or after `build`.

What should happen?

`gastby develop` on the repo above, should see 

1. console log "this is from /test.js"
1. an alert message

at http://localhost:8000/

### Actual result

none of the both happens.

### Environment

  System:
    OS: macOS Sierra 10.12.6
    CPU: x64 Intel(R) Core(TM) i5-4258U CPU @ 2.40GHz
    Shell: 5.2 - /bin/zsh
  Binaries:
    Node: 10.9.0 - /usr/local/bin/node
    Yarn: 1.9.4 - /usr/local/bin/yarn
    npm: 6.2.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 69.0.3497.100
    Firefox: 62.0
    Safari: 12.0
  npmPackages:
    gatsby: ^2.0.17 => 2.0.17 
    gatsby-plugin-manifest: ^2.0.4 => 2.0.4 
    gatsby-plugin-offline: ^2.0.5 => 2.0.5 
    gatsby-plugin-react-helmet: ^3.0.0 => 3.0.0 
  npmGlobalPackages:
    gatsby-cli: 2.4.2

**Comment by kakadiadarpan** [comment]
@github0013 I understand you are trying to load a script by a `<script />` tag, but why not just import the JS file directly?

**Comment by github0013** [comment]
Yes, I know I can just import the file, but there are conditions in my project that I have to use `<script/>` tag. 
I believe this issue must be a very minor case,  but I just found behavior difference between dev and prod that I thought worth mentioning.

Thanks

**Comment by cmawhorter** [comment]
i can't speak for OP but as someone looking to migrate an existing site to gatsby and doing some initial research, this issue stuck out.  

i'd like to copy/paste existing templates as much as possible.  at least as a first step.

this seems to be specifically for pages.  i'm not sure if components or html.js would have the same limitation?

**Comment by mufaddalmw** [comment]
@erw13n I tried importing as <script/> tag inside index.js and it threw error 
``` error  Parsing error: Unexpected token, expected ","
>  7 | <script src={withPrefix('../static/raphael')} />```

**Comment by ossd** [comment]
@erw13n the withPrefix did not solve the problem

When you build and serv the gatsby page, the script tags are in the code and the browser will load and execute them.
When you run develop, the script tags are in the code, too, but the browser will not load then. Interesting is, that the files are available, when you type them in the address bar of the browser.

But something in develop mode prevent the browser vom downloading and executing them. any ideas?

**Comment by ZachSaucier** [comment]
Facing the same issue as @ossd

**Comment by stanleyyuenyiu** [comment]
Facing the same issue too and withPredix did not work

**Comment by OverStruck** [comment]
Currently facing the same issue as well.
I need to add a JS script to a contact page so that the form gets processed by the script and sends the data to to my server.
Putting the script in the static folder and then adding `<script src={withPrefix("script.js")}></script>` in my contact page doesn't fire the JS inside the script.js file

**Comment by dpw1** [comment]
After 2 exhausting days I finally found a way to do it. 

1. Add your Javascript files to `static/js` 

2. Inside your `src` folder create a file named `html.js` and add the following to it:

```
import React from "react"
import PropTypes from "prop-types"
import { withPrefix } from "gatsby"

export default function HTML(props) {
  return (
    <html {...props.htmlAttributes}>
      <head>
        <meta charSet="utf-8" />
        <meta httpEquiv="x-ua-compatible" content="ie=edge" />
        <meta
          name="viewport"
          content="width=device-width, initial-scale=1, shrink-to-fit=no"
        />
        {props.headComponents}
      </head>
      <body {...props.bodyAttributes}>
        {props.preBodyComponents}
        <div
          key={`body`}
          id="___gatsby"
          dangerouslySetInnerHTML={{ __html: props.body }}
        />
        {props.postBodyComponents}

        {
          <link
            href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.9.0/css/all.min.css"
            rel="stylesheet"
          ></link>
        }

        <script
          type="text/javascript"
          src={withPrefix("js/vendors.js")}
        ></script>
      </body>
    </html>
  )
}

HTML.propTypes = {
  htmlAttributes: PropTypes.object,
  headComponents: PropTypes.array,
  bodyAttributes: PropTypes.object,
  preBodyComponents: PropTypes.array,
  body: PropTypes.string,
  postBodyComponents: PropTypes.array,
}
```

3. At the very bottom, before the `</body>` tag, notice the local script being imported from the `static` folder.

```
      <script
          type="text/javascript"
          src={withPrefix("js/vendors.js")}
        ></script>
```

I am also importing an external CSS file. Please notice that it's between curly brackets.

```
        {
          <link
            href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.9.0/css/all.min.css"
            rel="stylesheet"
          ></link>
        }
```
You can add a simple `alert("working")` in the script just to double check it is working.

From all the alternatives I found this was the only one that worked with me, I hope it helps you reading this as well.

**Comment by hasslarp** [comment]
The problem that <script>-tags aren't executed in development mode still exists for gatsby 3.14.4
Will this be resolved?

In my project I have to be able to fetch javascript code from source and inject it to the page. I know the security concerns, but in this case I need to secure the source instead.

Since the javascript is still executed on the final build I have to rebuild the project each time to test my code.
